lib/charms/vault_k8s/v0/vault_tls.py

```diff
@@ -342,23 +362,20 @@ def ca_certificate_set_in_peer_relation(self) -> bool:
             return False
         return False
 
-    """TODO: The following are methods that go through the container.
-    They will be extracted out in the future."""
-
     def _remove_all_certs_from_workload(self) -> None:
         """Removes the certificate files that are used for authentication."""
         self._remove_tls_file_from_workload(File.CA)
         self._remove_tls_file_from_workload(File.CERT)
         self._remove_tls_file_from_workload(File.CSR)
-        tlslogger.debug("Removed existing certificate files from workload.")
+        tls_logger.debug("Removed existing certificate files from workload.")
 
-    def _reload_vault_container(self) -> None:
+    def _reload_vault(self) -> None:
         """Sends a SIGHUP signal to the container which reloads Vault's files. Fails gracefully."""
         try:
-            self.charm._container.send_signal(SIGHUP, self.charm._container_name)
-            tlslogger.debug("Container restart requested")
+            self.substrate.send_signal(SIGHUP, "vault")
```

Comment:
> I still see no issue in using variables from the charm. That's the whole point of passing self as an argument to this class.
anything we access from the charm will be a requirement that we need to fulfil by the charm.
I prefer having the charm pass what it is required by lib in the constructor, defining a clear interface event if that means we will have a large init.
Passing self should be to access things like `charm.model` and `charm.app`.
At least that's how I have seen it done in other libs with a few exceptions. Of course we can have some exceptions here too as this lib is not expected to be used by charms other that `vault`, but I prefer to limit those exceptions to functions calls and not variables/constants.

Reply:
For now I will be passing this as a parameter in the constructor.
I'm happy to change it to use a getter function form the charm if that sounds better.

Reply:
> Passing self should be to access things like charm.model and charm.app

I don't see how this is any different. Why not pass `model` and `app` as init arguments also then? The container name is just as native of a value as `model` or `app` to the charm. If you want to absolutely guarantee that this value is not stale, you can locate the name of the container from the pebble layer too. A getter function just wraps this variable with extra lines, so I'd rather not do that.

I'm ok with passing it in the init. I'd like to prevent this seemingly unrelated function call from failing when the name of the container in the pebble layer changes when there's really no need for it to do that, even if I don't expect that to ever happen.